src/lib/metadataService.ts

```diff
@@ -0,0 +1,101 @@
+import { log } from 'console';
+
+type MetadataServiceInfoResponse = Record<
+  string,
+  {
+    name: string;
+    logo: string;
+    urls: {
+      website: string;
+      technical_doc: string;
+      cmc: string;
+    };
+    sector_tags: string[];
+    exchanges: string[];
+  }
+>;
+
+type MetadataServicePricesResponse = Record<
+  string,
+  {
+    price: number;
+    percent_change_24h: number;
+    volume_24h: number;
+    market_cap: number;
+  }
+>;
+
+type MetadataServiceCandlesResponse = Record<
+  string,
+  {
+    time: string; // ISO date string
+    open: number;
+    high: number;
+    low: number;
+    close: number;
+    volume: number;
+  }
+>;
+
+/**
+ * MetadataServiceClient
+ * @description client to fetch metadata info, prices, and candles
+ */
+class MetadataServiceClient {
+  private host: string;
+
+  constructor(host: string) {
+    if (!host) {
+      log('MetadataServiceClient requires a host');
+    }
+
+    this.host = host;
+  }
+
+  _post(endpoint: string, params: {} = {}) {
+    return fetch(`${this.host}/${endpoint}`, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify(params),
+    }).then((res) => res.json());
+  }
+
+  _get(endpoint: string) {
+    return fetch(`${this.host}/${endpoint}`).then((res) => res.json());
+  }
+
+  async getMarketmap(): Promise<Record<string, string>> {
+    return this._get('marketmap-map');
```

Comment:
[non blocking suggestion] convert string literals to str enums? i don't feel very opinionated about this one